### DISCUSSION TITLE
Throwing validation error only with stack trace

### DISCUSSION THREAD

**Discussion by nikola1912** [colinhacks/zod#963] [Q&A] [answered]
Hi all!

I'm trying to implement a `catch` block that would `console.error` the `stack` string from `ZodError` if the `error` is an instance of `ZodError`.  Otherwise it should log the whole `error` value. So I have something like this:
```ts
  try {
    const response = await storeAndValidate(data)
  } catch (error) {
    console.error(error instanceof ZodError ? error.stack : error)
  }
```

I want to avoid having to check if `error` is an instance of `ZodError` in order to log `error.stack`. My idea was to somehow make `z` throw `error.stack` directly. That way I wouldn't have to check `error instanceof ZodError` in every place I want to log the stack trace, I would simply write `console.error(error)` because I know that `error` would be the stack trace itself. 

So basically the code from above would become this and the resulting error logs would look the same:
```ts
  try {
    const response = await storeAndValidate(data)
  } catch (error) {
    console.error(error)
  }
```

Is this possible to implement?

**Comment by JacobWeisenburger** [comment, marked as answer]
Try this.

```ts
const logErrorStack = err => {
    console.error('stack' in err ? err.stack : err)
}
```
```ts
try {
    const response = await storeAndValidate(data)
} catch (error) {
    logErrorStack(error)
}
```
